designer/client/src/components/graph/node-modal/fragment-input-definition/item/Item.tsx

```diff
@@ -0,0 +1,84 @@
+import { Option } from "../FieldsSelect";
+import { FixedValuesPresets, ReturnedType, VariableTypes } from "../../../../../types";
+import SettingsButton from "../buttons/SettingsButton";
+import { FieldsRow } from "../FieldsRow";
+import { Settings } from "../settings/Settings";
+import { useDiffMark } from "../../PathsToMark";
+import { onChangeType, FragmentInputParameter } from "./";
+import { useFieldsContext } from "../NodeRowFields";
+
+interface ItemProps {
+    index: number;
+    item: FragmentInputParameter;
+    validators: Validator[];
+    namespace: string;
+    readOnly?: boolean;
+    showValidation?: boolean;
+    variableTypes: VariableTypes;
+    onChange: (path: string, value: onChangeType) => void;
+    options: Option[];
+    fixedValuesPresets: FixedValuesPresets;
+}
+
+export function Item(props: ItemProps): JSX.Element {
+    const { index, item, validators, namespace, variableTypes, readOnly, showValidation, onChange, options, fixedValuesPresets } = props;
+    const { getIsOpen, toggleIsOpen } = useFieldsContext();
+
+    const isOpen = getIsOpen(item.uuid);
+
+    const path = `${namespace}[${index}]`;
+    const [isMarked] = useDiffMark();
+    const getCurrentOption = useCallback(
+        (typ: ReturnedType | undefined) => {
+            const fallbackValue = { label: typ?.refClazzName, value: typ?.refClazzName };
+            const foundValue = options.find((item) => isEqual(typ?.refClazzName, item.value));
+            return foundValue || fallbackValue;
+        },
+        [options],
+    );
+
+    const openSettingMenu = () => {
+        toggleIsOpen(item.uuid);
+    };
+
+    return (
+        <div>
+            <FieldsRow index={index} uuid={item.uuid}>
+                <MapKey
+                    className="parametersFieldName"
+                    readOnly={readOnly}
+                    showValidation={showValidation}
+                    isMarked={isMarked(`${path}.name`)}
+                    onChange={(value) => onChange(`${path}.name`, value)}
+                    value={item.name}
+                    validators={validators}
+                />
+                <TypeSelect
+                    readOnly={readOnly}
+                    onChange={(value) => {
+                        onChange(`${path}.typ.refClazzName`, value);
+                    }}
+                    value={getCurrentOption(item.typ)}
+                    isMarked={isMarked(`${path}.typ.refClazzName`)}
+                    options={options}
+                />
+                <SettingsButton isOpen={isOpen} toggleIsOpen={openSettingMenu} />
+            </FieldsRow>
+            {isOpen && (
+                <Settings
+                    path={path}
+                    item={item}
+                    onChange={onChange}
+                    variableTypes={variableTypes}
+                    fixedValuesPresets={fixedValuesPresets}
+                    readOnly={readOnly}
+                />
+            )}
```

Comment:
We should have a way o displaying parameter config errors from BE. Right now the BE sends it with the fieldName = parameter name - we can either acknolowedge this as a TODO or adapt to current BE. In the future this should change to some convention like `$parameter.{parameterName}.{fieldName}` because right now if the user enters the parameter with name `$id` the errors will get displayed under the node name.